### PR TITLE
gh-120254: Add a `commands` argument to `pdb.set_trace`

### PR DESCRIPTION
**Pull request by gaogaotiantian**
* Issue: gh-120254

----
📚 Documentation preview 📚: https://cpython-previews--120255.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/pdb.rst [hunk 1]
```diff
@@ -211,6 +211,9 @@ access further features, you have to do this yourself:
    The *readrc* argument defaults to true and controls whether Pdb will load
    .pdbrc files from the filesystem.
 
+   The *commands* argument, if given, would be a list of commands to execute
+   when the debugger starts. It has similar effects to the :file:`.pdbrc` file.
```
Comment (picnixz) [inline Doc/library/pdb.rst]:
```suggestion
   The *commands* argument, if given, is a list of commands to execute
   when the debugger starts. Those commands are executed after those
   optionally provided by the :file:`.pdbrc` file. 
```

File: Misc/NEWS.d/next/Library/2024-06-08-03-29-01.gh-issue-120254.h682ke.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Added ``commands`` argument to :class:`pdb.Pdb` which allows user to send debugger commands in the source file.
```
Comment (picnixz) [inline Misc/NEWS.d/next/Library/2024-06-08-03-29-01.gh-issue-120254.h682ke.rst]:
```suggestion
Added ``commands`` argument to :class:`pdb.Pdb` which allows users to send debugger commands in the source file.
```

File: Doc/library/pdb.rst (hunk 1 above)
Comment (gaogaotiantian) [reply Doc/library/pdb.rst]: Do you think two "those" is a little bit confusing here? How about These commands are executed after any commands optionally provided by the :file:`.pdbrc` file?

Comment (mcepl) [inline Doc/library/pdb.rst]: The phrase “similar effects” is kind of vague. I am not sure how to say it more precisely. Link to the `.pdbrc` documentation? Not sure.

Comment (Delengowski) [reply Doc/library/pdb.rst]: I agree the double "those" is confusing.

Comment (picnixz) [reply Doc/library/pdb.rst]: Oups, you're right! I'm sorry I didn't reply to your question 3 weeks ago btw, but yes your suggestion is way better!

**Comment by gaogaotiantian** [comment]
I moved this argument to `set_trace()` where it makes more sense. Partially due to the impact of #121451 - we won't create a `Pdb` instance every time we try to bring up a debugger.

**Comment by gaogaotiantian** [comment]
Hi @brandtbucher , could you take a look at this feature and see if it's reasonable to add?

**Review by brandtbucher — APPROVED** [review]
Thanks a lot for your patience.

Finally got a chance to play with it... I loved finding out that `breakpoint(commands=[...])` works now! I think this is a great change.

**Comment by gaogaotiantian** [comment]
@brandtbucher do you think this needs a whatsnew entry?

**Comment by brandtbucher** [comment]
I think NEWS is probably good enough, but you can certainly add it to What's New if you feel it's something that you want to publicize more! Adding and removing stuff from that doc is pretty lightweight.

**Comment by gaogaotiantian** [comment]
I'll just merge this then!

**Pull request merged**